Shared/alerts/AlertConditionsController.h

```diff
@@ -48,6 +53,10 @@ class AlertConditionListModel;
 class AlertTarget;
 class LocationAlertSource;
 class LocationAlertTarget;
+class MessagesOverlay;
+
+using AlertTargetOptional = std::optional<AlertTarget*>;
```

Comment:
I revisited this and think it was done primarily to avoid having to create a typed pointer variable that was set to null in order to pass to QtFuture::makeReadyFuture which requires a type. But yes, returning null provided nearly the same usage after having reviewed it again. I think the optional looks a bit cleaner in a few places (has_value, init without typed pointer on another line). But I am not sure I could argue that it's entirely warranted.

Reply:
Reverted back to using QFuture<AlertTarget*>